src/content/1.7/modules/sample_modules/order_view_page_new_hooks/view-order-hooks-signature-widget.md

```diff
@@ -0,0 +1,448 @@
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table()
+ * @ORM\Entity(repositoryClass="PrestaShop\Module\DemoViewOrderHooks\Repository\OrderSignatureRepository")
+ */
+class OrderSignature
+{
+    /**
+     * @var int|null
+     *
+     * @ORM\Id
+     * @ORM\Column(name="id_signature", type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(name="id_order", type="integer")
+     */
+    private $orderId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(name="filename", type="string")
+     */
+    private $filename;
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return self
+     */
+    public function setId(int $id): self
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @return self
+     */
+    public function setFilename(string $filename): self
+    {
+        $this->filename = $filename;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+}
+
+```
+
+Let's put our signature picture, `john_doe.png` inside `/signatures/` folder.
+
+ {{< figure src="../../img/view-order-hooks-demo/john_doe.png" title="Signature" >}}
+
+Let's create Order Repository and data structures for interacting with Orders data.
+Data Transfer Object (https://en.wikipedia.org/wiki/Data_transfer_object)
+`Order.php` in `src/DTO` to carry data between processes. The main benefit is that it reduces
+the amount of data that needs to be sent inside application and encapsulates parameters for
+method calls (This can be useful if a method takes more than 4 or 5 parameters.). Also it is more 
+convenient than using associative array where you need to remember it's indexes to get the 
+certain values (no autocomplete is provided).
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\DTO;
+
+use DateTimeImmutable;
+
+final class Order
+{
+    /**
+     * @var int
+     */
+    private $orderId;
+
+    /**
+     * @var string
+     */
+    private $reference;
+
+    /**
+     * @var int
+     */
+    private $orderStateId;
+
+    /**
+     * @var DateTimeImmutable
+     */
+    private $orderDate;
+
+    public function __construct(int $orderId, string $reference, int $orderStateId, DateTimeImmutable $orderDate)
+    {
+        $this->orderId = $orderId;
+        $this->reference = $reference;
+        $this->orderStateId = $orderStateId;
```

Comment:
Why no getters for these properties? If they are not used maybe they should be removed?